src/lib/mina_generators/parties_generators.ml

```diff
@@ -1024,48 +1103,21 @@ let gen_parties_from ?(failure = None)
   (* table of public keys not in the ledger, to be used for new parties
      we have the corresponding private keys, so we can create signatures for those new parties
   *)
+  let ledger_account_set = Ledger.accounts ledger in
   let available_public_keys =
     let tbl = Signature_lib.Public_key.Compressed.Table.create () in
     Signature_lib.Public_key.Compressed.Map.iter_keys keymap ~f:(fun pk ->
         let account_id = Account_id.create pk Token_id.default in
-        if not (Account_id.Set.mem ledger_accounts account_id) then
+        if not (Account_id.Set.mem ledger_account_set account_id) then
```

Comment:
Since newly created accounts would also be recorded in `account_state_tbl`, should we also check against that one here?